src/explore-education-statistics-common/src/components/CopyLinkButton.tsx

```diff
@@ -1,51 +1,80 @@
 import Button from '@common/components/Button';
-import LinkIcon from '@common/components/LinkIcon';
 import styles from '@common/components/CopyLinkButton.module.scss';
-import Modal from '@common/components/Modal';
+import { CopyLinkModalProps } from '@common/components/CopyLinkModal';
 import UrlContainer from '@common/components/UrlContainer';
-import VisuallyHidden from '@common/components/VisuallyHidden';
 import useToggle from '@common/hooks/useToggle';
 import classNames from 'classnames';
-import React from 'react';
+import { OmitStrict } from '@common/types';
+import React, { ReactNode } from 'react';
 
-interface Props {
-  className?: string;
-  url: string;
+interface Props extends OmitStrict<CopyLinkModalProps, 'buttonClassName'> {
+  buttonText?: string | ReactNode;
+  label?: string;
 }
 
-const CopyLinkButton = ({ className, url }: Props) => {
+export default function CopyLinkButton({
+  buttonText = 'Copy',
+  className,
+  confirmMessage = 'Link copied to the clipboard.',
```

Comment:
As above comment - a more generic default would be 'Text copied to the clipboard' if we change this component to `CopyTextButton`.